### PR TITLE
[REG-653] master [ADD]  define qty to order from ca

### PR DESCRIPTION
**Pull request by romadjan**

### CODE REVIEW

File: addons/regency_shopsite/models/product_pricelist_item.py [hunk 1]
```diff
@@ -18,3 +18,51 @@ def _compute_name_and_price(self):
         for item in self:
             if item.overlay_tmpl_id and item.applied_on == '4_overlay_template':
                 item.name = "Shopsite Item Template: {}".format(item.overlay_tmpl_id.display_name)
+
+    def _is_applicable_for(self, product, qty_in_product_uom):
+        """Check whether the current rule is valid for the given product & qty.
+
```
Comment (chupaSV) [inline addons/regency_shopsite/models/product_pricelist_item.py:24]: Add a comment that it is an overriding of the core method.

File: addons/regency_shopsite/models/product_pricelist_item.py [hunk 2]
```diff
@@ -18,3 +18,51 @@ def _compute_name_and_price(self):
...
+                    and product.product_variant_id.id == self.product_id.id
+                ):
+                    # product self acceptable on template if has only one variant
+                    res = False
+            else:
+                if self.product_tmpl_id and product.product_tmpl_id.id != self.product_tmpl_id.id:
+                    res = False
+                elif self.product_id and product.id != self.product_id.id:
+                    res = False
+                elif self.overlay_tmpl_id and self.overlay_tmpl_id in product.overlay_template_ids:
```
Comment (chupaSV) [inline addons/regency_shopsite/models/product_pricelist_item.py:65]: Add comments before starting the custom code and after the code.  E.g:
`# START OF CUSTOM CODE`
`# END OF CUSTOM CODE`

**Review by chupaSV — CHANGES_REQUESTED** [review]

File: addons/consumption_agreement/models/consumption_agreement.py [hunk 3]
```diff
@@ -49,6 +50,17 @@ def _compute_tax_totals(self):
                 order.currency_id,
             )
 
+    @api.depends('line_ids', 'line_ids.qty_remaining', 'line_ids.product_uom_qty')
+    def _compute_hide_status(self):
+        for order in self:
+            if order:
+                if order.line_ids:
+                    hide = any(line > 0.0 for line in order.line_ids.mapped('qty_remaining'))
+                    if hide:
```
Comment (artemShelest) [inline addons/consumption_agreement/models/consumption_agreement.py:59]: `order.hide_create_so_button = not hide`

File: addons/consumption_agreement/static/src/js/consumption_agreement_portal.js [hunk 4]
```diff
@@ -87,7 +89,60 @@ odoo.define('consumption_agreement.consumption_agreement_portal', function (requ
...
+        let self = this,
+            $target = $(ev.currentTarget),
+            quantity = parseInt($target.val());
+
+        this._callUpdateLineRoute(self.orderDetail.orderId, {
+            'line_id': $target.data('lineId'),
+            'input_quantity': quantity >= 0 ? quantity : false,
+            'access_token': self.orderDetail.token
+        }).then((data) => {
+            console.log( $target.closest('tr').find('.js_agreement_line_quantity').val(data.agreement_line_id))
```
Comment (artemShelest) [inline addons/consumption_agreement/static/src/js/consumption_agreement_portal.js:118]: Handle errors from the controller (when it returns False). Better change False to something readable and meaningful.
Same for the _callUpdateLineRoute() call below.

**Review by artemShelest — CHANGES_REQUESTED** [review]

**Review by chupaSV — APPROVED** [review]

**Review by artemShelest — APPROVED** [review]

**Pull request merged**